lib/screens/preferences_screen.dart

```diff
@@ -0,0 +1,180 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+import '../services/language_service.dart';
+import '../models/language_model.dart';
+import 'error_screen.dart';
+import 'sentence_list_screen.dart';
+
+class PreferenceScreen extends StatefulWidget {
+  @override
+  State createState() => PreferenceScreenState();
+}
+
+class PreferenceScreenState extends State<PreferenceScreen> {
+  String _preferredSourceLang;
+  String _preferredTargetLang;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  Widget _loadDropDownItems(List<Language> data, String languageType) {
+    String hint =
+        (languageType == "source") ? "Source Language" : "Target Language";
```

Comment:
I am always nervous when doing string comparisons like this. It is so fragile. At the very least convert languageType to lower case before comparing. Even better languageType could be an enum, I understand that it might not be the good choice for what you are trying to achieve. So let me know your thoughts.